### DISCUSSION TITLE
Why doesn't this repo have more stars?!

### DISCUSSION THREAD

**Discussion by hirasso** [sniccowp/sniccowp#201] [General]
Hi there @calvinalkan , 

Yesterday I was researching a WordPress router to replace the `add_rewrite_rules` madness. Luckily, I was using the GitHub iOS apps' code search, and low and behold, I stumbled upon snicco! 

![signal-2024-08-05-105601](https://github.com/user-attachments/assets/db40e60d-6a8a-43d4-8175-1b60b0ee59cf)

I'm absolutely blown away by, like, _everything_ I'm seeing in this repo. And right now very motivated to start using it throughout my future projects. The thing I was pretty surprised about was that snicco doesn't have more stars. And I think it should have more of these 🌟

So I'm starting this discussion to research ways to make snicco more known in the development community. First ideas:

- Create a logo
- Build a documentation website
- Post on hacker news

I might be able to help with parts of that stuff (I'm a maintainer of https://swup.js.org and built their docs website) – but I think the first step is to acknowledge that snicco needs some noise :)

**Comment by calvinalkan** [comment]
Hey @hirasso 

> I'm absolutely blown away by, like, everything I'm seeing in this repo. And right now very motivated to start using it throughout my future projects.

Thanks for the kind words.

> The thing I was pretty surprised about was that snicco doesn't have more stars

This might be two-fold. 

a) Many of the packages in here are internal dependencies that we use for our product [Fortress](https://fortress.snicco.io/), so the # of stars has historically not been the primary way we measure "success" of this project, nor did we try to create the "next popular open-source thing."
This project came about because I faced many problems developing what I consider high-quality, maintainable, and testable software in the WordPress space.

Examples:

- > router to replace the add_rewrite_rules madness
- [DBAL](https://github.com/snicco/better-wpdb)
- [WP-CLI reliability](https://github.com/snicco/better-wp-cli)


b) I think most developers in the WordPress space don't even perceive about the problems we solved here. This is probably because most WordPress software is the opposite of our goals: untested, unreliable, and a pain to maintain long term.

This meme comes to mind when I tried to explain the purpose of this project at times:

![image](https://github.com/user-attachments/assets/4965b5bf-0539-4eeb-84f9-c5d174a4e328)

---

> So I'm starting this discussion to research ways to make snicco more known in the development community. First ideas:
>
>    Create a logo
>    Build a documentation website
>    Post on hacker news

I'm all for making this more popular and "raising the bar" in the WordPress dev community, but it's not the main focus of our company right now.

The next steps as far as `snicco/snicco`
are:

- Tagging a v2, that incorporates many of our learnings from using it in production in Fortress.
- Building a small website & documentation hub under our company site.
- Giving the framework a proper name,
- etc.

While that's happening, you're of course more than welcome to use the project's components in your projects!

I'm happy you find it useful.
